### ISSUE TITLE
04_pytorch_custom_datasets.ipynb: .detach() required by matplotlib when plotting loss curves

### ISSUE THREAD

**Issue by cswamy** [mrdbourke/pytorch-deep-learning#583]
Hello,

**Video**: `04_pytorch_custom_datasets.ipynb`

**Issue**: After building and training baseline model on CPU (GPUs were not available on colab), when I tried to plot loss curves with `model_0_results` dict using function `plot_loss_curves`, got a RuntimeError in colab `Can't call numpy() on Tensor that requires grad. Use tensor.detach().numpy() instead.`

**Investigation**: Had a look at the values in results dictionary with `model_0_results.values()`. I could see `results["train_loss"]` contained list of tensors with gradient (assuming from training step). 

**Fix**: I could get it to work by going back to the `train` function and changing `results["train_loss"].append(train_loss)` to `results["train_loss"].append(train_loss.detach())`

Not sure if something has changed since course was recorded but thought I would mention this in case it's useful for others. Loving this course - thanks for putting it together!

**Comment by cswamy** [comment]
Further digging realised this won't be an issue using `item()` when calculating loss and accuracy. I had missed the `item()` in my train and test functions. Closing issue.
